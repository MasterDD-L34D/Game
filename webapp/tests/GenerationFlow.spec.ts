import { spawnSync } from 'node:child_process';
import path from 'node:path';
import { fileURLToPath } from 'node:url';

import { mount } from '@vue/test-utils';
import { describe, expect, it } from 'vitest';

import SpeciesPanel from '../src/components/SpeciesPanel.vue';

const __filename = fileURLToPath(import.meta.url);
const __dirname = path.dirname(__filename);

function orchestrateSpecies() {
  const repoRoot = path.resolve(__dirname, '..', '..');
  const scriptPath = path.resolve(repoRoot, 'services', 'generation', 'orchestrator.py');
  const payload = {
    trait_ids: ['artigli_sette_vie', 'coda_frusta_cinetica', 'scheletro_idro_regolante'],
    seed: 321,
    base_name: 'Predatore Snapshot',
    biome_id: 'caverna_risonante',
  };
  const pythonExecutable = process.env.PYTHON || 'python3';
  const result = spawnSync(pythonExecutable, [scriptPath, '--action', 'generate-species'], {
    cwd: repoRoot,
    input: JSON.stringify(payload),
    encoding: 'utf8',
  });
  if (result.status !== 0) {
    throw new Error(result.stderr || `orchestrator exited with code ${result.status}`);
  }
  return JSON.parse(result.stdout);
}

function orchestrateSpeciesBatch() {
  const repoRoot = path.resolve(__dirname, '..', '..');
  const scriptPath = path.resolve(repoRoot, 'services', 'generation', 'orchestrator.py');
  const payload = {
    batch: [
      {
        trait_ids: ['artigli_sette_vie', 'coda_frusta_cinetica', 'scheletro_idro_regolante'],
        seed: 321,
        base_name: 'Predatore Snapshot',
        biome_id: 'caverna_risonante',
      },
      {
        trait_ids: ['artigli_sette_vie', 'coda_frusta_cinetica'],
        seed: 322,
        base_name: 'Predatore Variant',
        biome_id: 'caverna_risonante',
        fallback_trait_ids: ['scheletro_idro_regolante'],
      },
    ],
  };
  const pythonExecutable = process.env.PYTHON || 'python3';
  const result = spawnSync(pythonExecutable, [scriptPath, '--action', 'generate-species-batch'], {
    cwd: repoRoot,
    input: JSON.stringify(payload),
    encoding: 'utf8',
  });
  if (result.status !== 0) {
    throw new Error(result.stderr || `orchestrator exited with code ${result.status}`);
  }
  return JSON.parse(result.stdout);
}

describe('Generation flow orchestrato', () => {
  it('renderizza il blueprint orchestrato', async () => {
    const response = orchestrateSpecies();
    const batch = orchestrateSpeciesBatch();
    const wrapper = mount(SpeciesPanel, {
      props: {
        species: response.blueprint,
        validation: response.validation,
        meta: response.meta,
        previewBatch: batch.results,
        autoPreview: false,
      },
    });

    expect(wrapper.text()).toContain('Predatore Snapshot');
    expect(response.meta.fallback_used).toBe(false);
    expect(Array.isArray(response.validation.messages)).toBe(true);

    const synergyTab = wrapper
      .findAll('button')
      .find((button) => button.text().toLowerCase().includes('sinergie'));
    expect(synergyTab).toBeDefined();
    if (synergyTab) {
      await synergyTab.trigger('click');
    }

    expect(wrapper.text()).toContain('Anteprime sintetiche');
    expect(wrapper.html()).toMatchSnapshot();
  });
});
