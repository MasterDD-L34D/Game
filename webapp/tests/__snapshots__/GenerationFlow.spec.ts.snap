// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Generation flow orchestrato > renderizza il blueprint orchestrato 1`] = `
"<section data-v-73eb6eb5="" class="species-panel">
  <section data-v-966c492d="" data-v-73eb6eb5="" class="nebula-shell">
    <div data-v-966c492d="" class="nebula-shell__frame">
      <header data-v-966c492d="" class="nebula-shell__header">
        <div data-v-966c492d="" class="nebula-shell__status-grid">
          <div data-v-966c492d="" class="nebula-shell__status"><span data-v-966c492d="" class="nebula-shell__status-led" data-tone="neutral"></span>
            <div data-v-966c492d="">
              <p data-v-966c492d="" class="nebula-shell__status-label">Rarità</p>
              <p data-v-966c492d="" class="nebula-shell__status-value">R1</p>
            </div>
          </div>
          <div data-v-966c492d="" class="nebula-shell__status"><span data-v-966c492d="" class="nebula-shell__status-led" data-tone="neutral"></span>
            <div data-v-966c492d="">
              <p data-v-966c492d="" class="nebula-shell__status-label">Threat tier</p>
              <p data-v-966c492d="" class="nebula-shell__status-value">T1</p>
            </div>
          </div>
          <div data-v-966c492d="" class="nebula-shell__status"><span data-v-966c492d="" class="nebula-shell__status-led" data-tone="critical"></span>
            <div data-v-966c492d="">
              <p data-v-966c492d="" class="nebula-shell__status-label">Allineamento</p>
              <p data-v-966c492d="" class="nebula-shell__status-value">33%</p>
            </div>
          </div>
        </div>
        <div data-v-966c492d="" class="nebula-shell__actions">
          <div data-v-78ecd3ce="" data-v-73eb6eb5="" class="species-quick-actions"><button data-v-78ecd3ce="" type="button" class="species-quick-actions__button"> Esporta scheda </button><button data-v-78ecd3ce="" type="button" class="species-quick-actions__button species-quick-actions__button--secondary"> Salva nel pack </button></div>
        </div>
      </header>
      <div data-v-966c492d="" class="nebula-shell__cards">
        <article data-v-1960f20b="" data-v-73eb6eb5="" class="species-card">
          <header data-v-1960f20b="" class="species-card__header">
            <div data-v-1960f20b="">
              <p data-v-1960f20b="" class="species-card__kicker">Profilo non definito</p>
              <h3 data-v-1960f20b="" class="species-card__title">Predatore Snapshot / Scheletro Idro-Regolante</h3>
            </div>
            <div data-v-1960f20b="" class="species-card__badges"><span data-v-48844140="" data-v-1960f20b="" class="trait-chip" data-variant="core"><span data-v-48844140="" class="trait-chip__icon" aria-hidden="true">✦</span><span data-v-48844140="" class="trait-chip__label">R1</span></span><span data-v-48844140="" data-v-1960f20b="" class="trait-chip" data-variant="hazard"><span data-v-48844140="" class="trait-chip__icon" aria-hidden="true">⚡</span><span data-v-48844140="" class="trait-chip__label">T1</span></span></div>
          </header>
          <p data-v-1960f20b="" class="species-card__synopsis">Artigli a Sette Vie, Coda a Frusta Cinetica, Scheletro Idro-Regolante</p>
          <section data-v-1960f20b="" class="species-card__traits" aria-label="Tratti principali">
            <div data-v-1960f20b="">
              <h4 data-v-1960f20b="">Core</h4>
              <div data-v-1960f20b="" class="species-card__trait-grid"><span data-v-48844140="" data-v-1960f20b="" class="trait-chip" data-variant="core"><span data-v-48844140="" class="trait-chip__icon" aria-hidden="true">✶</span><span data-v-48844140="" class="trait-chip__label">artigli_sette_vie</span></span><span data-v-48844140="" data-v-1960f20b="" class="trait-chip" data-variant="core"><span data-v-48844140="" class="trait-chip__icon" aria-hidden="true">✶</span><span data-v-48844140="" class="trait-chip__label">coda_frusta_cinetica</span></span><span data-v-48844140="" data-v-1960f20b="" class="trait-chip" data-variant="core"><span data-v-48844140="" class="trait-chip__icon" aria-hidden="true">✶</span><span data-v-48844140="" class="trait-chip__label">scheletro_idro_regolante</span></span></div>
            </div>
            <div data-v-1960f20b="">
              <h4 data-v-1960f20b="">Derivati</h4>
              <div data-v-1960f20b="" class="species-card__trait-grid"><span data-v-48844140="" data-v-1960f20b="" class="trait-chip" data-variant="derived"><span data-v-48844140="" class="trait-chip__icon" aria-hidden="true">☲</span><span data-v-48844140="" class="trait-chip__label">ali_ioniche</span></span><span data-v-48844140="" data-v-1960f20b="" class="trait-chip" data-variant="derived"><span data-v-48844140="" class="trait-chip__icon" aria-hidden="true">☲</span><span data-v-48844140="" class="trait-chip__label">antenne_flusso_mareale</span></span><span data-v-48844140="" data-v-1960f20b="" class="trait-chip" data-variant="derived"><span data-v-48844140="" class="trait-chip__icon" aria-hidden="true">☲</span><span data-v-48844140="" class="trait-chip__label">antenne_tesla</span></span><span data-v-48844140="" data-v-1960f20b="" class="trait-chip" data-variant="derived"><span data-v-48844140="" class="trait-chip__icon" aria-hidden="true">☲</span><span data-v-48844140="" class="trait-chip__label">antenne_wideband</span></span><span data-v-48844140="" data-v-1960f20b="" class="trait-chip" data-variant="derived"><span data-v-48844140="" class="trait-chip__icon" aria-hidden="true">☲</span><span data-v-48844140="" class="trait-chip__label">artigli_induzione</span></span><span data-v-48844140="" data-v-1960f20b="" class="trait-chip" data-variant="derived"><span data-v-48844140="" class="trait-chip__icon" aria-hidden="true">☲</span><span data-v-48844140="" class="trait-chip__label">artigli_sette_vie</span></span><span data-v-48844140="" data-v-1960f20b="" class="trait-chip" data-variant="derived"><span data-v-48844140="" class="trait-chip__icon" aria-hidden="true">☲</span><span data-v-48844140="" class="trait-chip__label">artigli_vetrificati</span></span><span data-v-48844140="" data-v-1960f20b="" class="trait-chip" data-variant="derived"><span data-v-48844140="" class="trait-chip__icon" aria-hidden="true">☲</span><span data-v-48844140="" class="trait-chip__label">barbigli_sensori_plasma</span></span><span data-v-48844140="" data-v-1960f20b="" class="trait-chip" data-variant="derived"><span data-v-48844140="" class="trait-chip__icon" aria-hidden="true">☲</span><span data-v-48844140="" class="trait-chip__label">biochip_memoria</span></span><span data-v-48844140="" data-v-1960f20b="" class="trait-chip" data-variant="derived"><span data-v-48844140="" class="trait-chip__icon" aria-hidden="true">☲</span><span data-v-48844140="" class="trait-chip__label">branchie_dual_mode</span></span><span data-v-48844140="" data-v-1960f20b="" class="trait-chip" data-variant="derived"><span data-v-48844140="" class="trait-chip__icon" aria-hidden="true">☲</span><span data-v-48844140="" class="trait-chip__label">branchie_metalloidi</span></span><span data-v-48844140="" data-v-1960f20b="" class="trait-chip" data-variant="derived"><span data-v-48844140="" class="trait-chip__icon" aria-hidden="true">☲</span><span data-v-48844140="" class="trait-chip__label">branchie_osmotiche_salmastra</span></span><span data-v-48844140="" data-v-1960f20b="" class="trait-chip" data-variant="derived"><span data-v-48844140="" class="trait-chip__icon" aria-hidden="true">☲</span><span data-v-48844140="" class="trait-chip__label">camere_anticorrosione</span></span><span data-v-48844140="" data-v-1960f20b="" class="trait-chip" data-variant="derived"><span data-v-48844140="" class="trait-chip__icon" aria-hidden="true">☲</span><span data-v-48844140="" class="trait-chip__label">capillari_criogenici</span></span><span data-v-48844140="" data-v-1960f20b="" class="trait-chip" data-variant="derived"><span data-v-48844140="" class="trait-chip__icon" aria-hidden="true">☲</span><span data-v-48844140="" class="trait-chip__label">capillari_fotovoltaici</span></span><span data-v-48844140="" data-v-1960f20b="" class="trait-chip" data-variant="derived"><span data-v-48844140="" class="trait-chip__icon" aria-hidden="true">☲</span><span data-v-48844140="" class="trait-chip__label">cartilagini_biofibre</span></span><span data-v-48844140="" data-v-1960f20b="" class="trait-chip" data-variant="derived"><span data-v-48844140="" class="trait-chip__icon" aria-hidden="true">☲</span><span data-v-48844140="" class="trait-chip__label">cartilagini_pseudometalliche</span></span><span data-v-48844140="" data-v-1960f20b="" class="trait-chip" data-variant="derived"><span data-v-48844140="" class="trait-chip__icon" aria-hidden="true">☲</span><span data-v-48844140="" class="trait-chip__label">chemiorecettori_bromuro</span></span><span data-v-48844140="" data-v-1960f20b="" class="trait-chip" data-variant="derived"><span data-v-48844140="" class="trait-chip__icon" aria-hidden="true">☲</span><span data-v-48844140="" class="trait-chip__label">circolazione_supercritica</span></span><span data-v-48844140="" data-v-1960f20b="" class="trait-chip" data-variant="derived"><span data-v-48844140="" class="trait-chip__icon" aria-hidden="true">☲</span><span data-v-48844140="" class="trait-chip__label">cisti_iperbariche</span></span><span data-v-48844140="" data-v-1960f20b="" class="trait-chip" data-variant="derived"><span data-v-48844140="" class="trait-chip__icon" aria-hidden="true">☲</span><span data-v-48844140="" class="trait-chip__label">coda_contrappeso</span></span><span data-v-48844140="" data-v-1960f20b="" class="trait-chip" data-variant="derived"><span data-v-48844140="" class="trait-chip__icon" aria-hidden="true">☲</span><span data-v-48844140="" class="trait-chip__label">coda_frusta_cinetica</span></span><span data-v-48844140="" data-v-1960f20b="" class="trait-chip" data-variant="derived"><span data-v-48844140="" class="trait-chip__icon" aria-hidden="true">☲</span><span data-v-48844140="" class="trait-chip__label">coda_stabilizzatrice_vortex</span></span><span data-v-48844140="" data-v-1960f20b="" class="trait-chip" data-variant="derived"><span data-v-48844140="" class="trait-chip__icon" aria-hidden="true">☲</span><span data-v-48844140="" class="trait-chip__label">cromofori_alert_acido</span></span><span data-v-48844140="" data-v-1960f20b="" class="trait-chip" data-variant="derived"><span data-v-48844140="" class="trait-chip__icon" aria-hidden="true">☲</span><span data-v-48844140="" class="trait-chip__label">cuscinetti_elettrostatici</span></span><span data-v-48844140="" data-v-1960f20b="" class="trait-chip" data-variant="derived"><span data-v-48844140="" class="trait-chip__icon" aria-hidden="true">☲</span><span data-v-48844140="" class="trait-chip__label">denti_chelatanti</span></span><span data-v-48844140="" data-v-1960f20b="" class="trait-chip" data-variant="derived"><span data-v-48844140="" class="trait-chip__icon" aria-hidden="true">☲</span><span data-v-48844140="" class="trait-chip__label">denti_tuning_fork</span></span><span data-v-48844140="" data-v-1960f20b="" class="trait-chip" data-variant="derived"><span data-v-48844140="" class="trait-chip__icon" aria-hidden="true">☲</span><span data-v-48844140="" class="trait-chip__label">enzimi_antifase_termica</span></span><span data-v-48844140="" data-v-1960f20b="" class="trait-chip" data-variant="derived"><span data-v-48844140="" class="trait-chip__icon" aria-hidden="true">☲</span><span data-v-48844140="" class="trait-chip__label">enzimi_metanoossidanti</span></span><span data-v-48844140="" data-v-1960f20b="" class="trait-chip" data-variant="derived"><span data-v-48844140="" class="trait-chip__icon" aria-hidden="true">☲</span><span data-v-48844140="" class="trait-chip__label">filamenti_magnetotrofi</span></span><span data-v-48844140="" data-v-1960f20b="" class="trait-chip" data-variant="derived"><span data-v-48844140="" class="trait-chip__icon" aria-hidden="true">☲</span><span data-v-48844140="" class="trait-chip__label">filamenti_termoconduzione</span></span><span data-v-48844140="" data-v-1960f20b="" class="trait-chip" data-variant="derived"><span data-v-48844140="" class="trait-chip__icon" aria-hidden="true">☲</span><span data-v-48844140="" class="trait-chip__label">foliaggio_spugna</span></span><span data-v-48844140="" data-v-1960f20b="" class="trait-chip" data-variant="derived"><span data-v-48844140="" class="trait-chip__icon" aria-hidden="true">☲</span><span data-v-48844140="" class="trait-chip__label">ghiandole_condensa_ozono</span></span><span data-v-48844140="" data-v-1960f20b="" class="trait-chip" data-variant="derived"><span data-v-48844140="" class="trait-chip__icon" aria-hidden="true">☲</span><span data-v-48844140="" class="trait-chip__label">ghiandole_fango_calde</span></span><span data-v-48844140="" data-v-1960f20b="" class="trait-chip" data-variant="derived"><span data-v-48844140="" class="trait-chip__icon" aria-hidden="true">☲</span><span data-v-48844140="" class="trait-chip__label">ghiandole_iodoattive</span></span><span data-v-48844140="" data-v-1960f20b="" class="trait-chip" data-variant="derived"><span data-v-48844140="" class="trait-chip__icon" aria-hidden="true">☲</span><span data-v-48844140="" class="trait-chip__label">ghiandole_nebbia_ionica</span></span><span data-v-48844140="" data-v-1960f20b="" class="trait-chip" data-variant="derived"><span data-v-48844140="" class="trait-chip__icon" aria-hidden="true">☲</span><span data-v-48844140="" class="trait-chip__label">ghiandole_ventosa</span></span><span data-v-48844140="" data-v-1960f20b="" class="trait-chip" data-variant="derived"><span data-v-48844140="" class="trait-chip__icon" aria-hidden="true">☲</span><span data-v-48844140="" class="trait-chip__label">gusci_magnesio</span></span><span data-v-48844140="" data-v-1960f20b="" class="trait-chip" data-variant="derived"><span data-v-48844140="" class="trait-chip__icon" aria-hidden="true">☲</span><span data-v-48844140="" class="trait-chip__label">lamine_filtranti_aeree</span></span><span data-v-48844140="" data-v-1960f20b="" class="trait-chip" data-variant="derived"><span data-v-48844140="" class="trait-chip__icon" aria-hidden="true">☲</span><span data-v-48844140="" class="trait-chip__label">linfa_tampone</span></span><span data-v-48844140="" data-v-1960f20b="" class="trait-chip" data-variant="derived"><span data-v-48844140="" class="trait-chip__icon" aria-hidden="true">☲</span><span data-v-48844140="" class="trait-chip__label">mantelli_geotermici</span></span><span data-v-48844140="" data-v-1960f20b="" class="trait-chip" data-variant="derived"><span data-v-48844140="" class="trait-chip__icon" aria-hidden="true">☲</span><span data-v-48844140="" class="trait-chip__label">membrane_pneumatofori</span></span><span data-v-48844140="" data-v-1960f20b="" class="trait-chip" data-variant="derived"><span data-v-48844140="" class="trait-chip__icon" aria-hidden="true">☲</span><span data-v-48844140="" class="trait-chip__label">mimetismo_cromatico_passivo</span></span><span data-v-48844140="" data-v-1960f20b="" class="trait-chip" data-variant="derived"><span data-v-48844140="" class="trait-chip__icon" aria-hidden="true">☲</span><span data-v-48844140="" class="trait-chip__label">mucose_aderenza_sonica</span></span><span data-v-48844140="" data-v-1960f20b="" class="trait-chip" data-variant="derived"><span data-v-48844140="" class="trait-chip__icon" aria-hidden="true">☲</span><span data-v-48844140="" class="trait-chip__label">sacche_galleggianti_ascensoriali</span></span><span data-v-48844140="" data-v-1960f20b="" class="trait-chip" data-variant="derived"><span data-v-48844140="" class="trait-chip__icon" aria-hidden="true">☲</span><span data-v-48844140="" class="trait-chip__label">struttura_elastica_amorfa</span></span><span data-v-48844140="" data-v-1960f20b="" class="trait-chip" data-variant="derived"><span data-v-48844140="" class="trait-chip__icon" aria-hidden="true">☲</span><span data-v-48844140="" class="trait-chip__label">tattiche_di_branco</span></span></div>
            </div>
            <!--v-if-->
          </section>
          <footer data-v-1960f20b="" class="species-card__footer">
            <div data-v-1960f20b="" class="species-card__info-block"><span data-v-1960f20b="" class="species-card__label">Energia</span><span data-v-1960f20b="">medio</span></div>
            <div data-v-1960f20b="" class="species-card__info-block"><span data-v-1960f20b="" class="species-card__label">Habitat</span><span data-v-1960f20b="">caverna_risonante, falde_magnetiche_psioniche, orbita_psionica_inversa</span></div>
            <!--v-if-->
          </footer>
        </article>
      </div>
      <nav data-v-966c492d="" class="nebula-shell__tabs" aria-label="Navigazione schede" role="tablist"><button data-v-966c492d="" type="button" class="nebula-shell__tab" role="tab" id="nebula-shell-1-tab-0" aria-controls="nebula-shell-1-panel-0" aria-selected="false" tabindex="-1"><span data-v-966c492d="" class="nebula-shell__tab-icon" aria-hidden="true">🧬</span><span data-v-966c492d="">Scheda</span></button><button data-v-966c492d="" type="button" class="nebula-shell__tab" role="tab" id="nebula-shell-1-tab-1" aria-controls="nebula-shell-1-panel-1" aria-selected="false" tabindex="-1"><span data-v-966c492d="" class="nebula-shell__tab-icon" aria-hidden="true">🌿</span><span data-v-966c492d="">Biologia</span></button><button data-v-966c492d="" type="button" class="nebula-shell__tab" role="tab" id="nebula-shell-1-tab-2" aria-controls="nebula-shell-1-panel-2" aria-selected="false" tabindex="-1"><span data-v-966c492d="" class="nebula-shell__tab-icon" aria-hidden="true">📡</span><span data-v-966c492d="">Telemetry</span></button><button data-v-966c492d="" type="button" class="nebula-shell__tab nebula-shell__tab--active" role="tab" id="nebula-shell-1-tab-3" aria-controls="nebula-shell-1-panel-3" aria-selected="true" tabindex="0"><span data-v-966c492d="" class="nebula-shell__tab-icon" aria-hidden="true">∞</span><span data-v-966c492d="">Sinergie</span></button></nav>
      <div data-v-966c492d="" class="nebula-shell__content" role="tabpanel" id="nebula-shell-1-panel-3" aria-labelledby="nebula-shell-1-tab-3" tabindex="0">
        <div data-v-73eb6eb5="" class="species-panel__section species-panel__section--synergy">
          <p data-v-73eb6eb5="" class="species-panel__empty">Nessuna sinergia registrata per questa specie.</p>
          <section data-v-38b0498e="" data-v-73eb6eb5="" class="species-preview-grid species-panel__previews" aria-busy="false">
            <header data-v-38b0498e="" class="species-preview-grid__header">
              <h3 data-v-38b0498e="">Anteprime sintetiche</h3><button data-v-73eb6eb5="" type="button" class="species-panel__refresh"> Aggiorna batch </button>
            </header>
            <div data-v-38b0498e="" class="species-preview-grid__cards" role="list">
              <article data-v-38b0498e="" class="species-preview-card" role="listitem">
                <header data-v-38b0498e="" class="species-preview-card__header">
                  <h4 data-v-38b0498e="">Predatore Snapshot / Scheletro Idro-Regolante</h4><span data-v-38b0498e="" class="species-preview-card__badge">T1</span>
                </header>
                <p data-v-38b0498e="" class="species-preview-card__summary">Artigli a Sette Vie, Coda a Frusta Cinetica, Scheletro Idro-Regolante</p>
                <dl data-v-38b0498e="" class="species-preview-card__stats">
                  <div data-v-38b0498e="">
                    <dt data-v-38b0498e="">Energia</dt>
                    <dd data-v-38b0498e="">medio</dd>
                  </div>
                  <div data-v-38b0498e="">
                    <dt data-v-38b0498e="">Rarità</dt>
                    <dd data-v-38b0498e="">R1</dd>
                  </div>
                </dl>
                <p data-v-38b0498e="" class="species-preview-card__traits">artigli_sette_vie, coda_frusta_cinetica, scheletro_idro_regolante</p>
              </article>
              <article data-v-38b0498e="" class="species-preview-card" role="listitem">
                <header data-v-38b0498e="" class="species-preview-card__header">
                  <h4 data-v-38b0498e="">Predatore Variant / Coda a Frusta Cinetica</h4><span data-v-38b0498e="" class="species-preview-card__badge">T1</span>
                </header>
                <p data-v-38b0498e="" class="species-preview-card__summary">Artigli a Sette Vie, Coda a Frusta Cinetica</p>
                <dl data-v-38b0498e="" class="species-preview-card__stats">
                  <div data-v-38b0498e="">
                    <dt data-v-38b0498e="">Energia</dt>
                    <dd data-v-38b0498e="">medio</dd>
                  </div>
                  <div data-v-38b0498e="">
                    <dt data-v-38b0498e="">Rarità</dt>
                    <dd data-v-38b0498e="">R1</dd>
                  </div>
                </dl>
                <p data-v-38b0498e="" class="species-preview-card__traits">artigli_sette_vie, coda_frusta_cinetica</p>
              </article>
            </div>
          </section>
        </div>
      </div>
      <footer data-v-966c492d="" class="nebula-shell__footer"></footer>
    </div>
  </section>
</section>"
`;
